core/domain/caching_services.py

```diff
@@ -98,8 +100,8 @@
     CACHE_NAMESPACE_STORY: lambda x: x.serialize(),
     CACHE_NAMESPACE_TOPIC: lambda x: x.serialize(),
     CACHE_NAMESPACE_PLATFORM_PARAMETER: lambda x: x.serialize(),
```

Comment:
Could you check whether we need to .encode('utf-8') for all the other items as well, or whether we just haven't seen this yet because we haven't tested with unicode chars?

Also, have stronger, comprehensive backend tests for all of this. Make sure they include unicode cases as well, and a full range of different types.

Reply:
For the unicode test cases, try to have the tests mimic exactly what happens if a similar character is saved in a topic/exploration/etc. html/text field.

Reply:
Sorry I'm still have a hard time understanding exactly what you mean. Currently I have a test that saves a similar string containing a unicode character(emoji) to the backend fully utilizing the serialization and deserialization processing in the caching_services class. Each of the tests are then made to test if the storage and retrieval does not alter the emoji. 

Is this what you mean or am i understanding it incorrectly.

Reply:
Ah -- to clarify, what you really want to do is to make sure that no errors show up when we run this in the main server, which contains historical unicode data. So, we need to make sure that the test env is in sync with the main server, so that if the tests pass, we can be sure that prod will work fine too.

So I would suggest:

(a) Spin up a local dev server. Save a config property that has unicode chars, using the local dev server's GUI. Write a piece of logging code for the type/contents of the stored field and print it to the terminal.

(b) In tests, do the same thing programmatically, and write a similar piece of logging code. Verify that what is printed to the terminal is the same as that in (a). If it isn't, update your test until it is.

Do the same for an exploration that has unicode chars all over the place. These checks help to validate that what you are testing is encoded similarly to existing data.

Does that help?

Reply:
I think so! I added the tests for exploration and config properties.